tests/plugins/test_koji_upload.py

```diff
@@ -187,27 +184,41 @@ def taskFinished(self, task_id):
         return self.tag_task_state in ['CLOSED', 'FAILED', 'CANCELED', None]
 
 
-FAKE_SIGMD5 = b'0' * 32
-FAKE_RPM_OUTPUT = (
-    b'name1;1.0;1;' + B_LOCAL_ARCH + b';0;' + FAKE_SIGMD5 + b';(none);'
-    b'RSA/SHA256, Mon 29 Jun 2015 13:58:22 BST, Key ID abcdef01234567\n'
+class MockedRpmHeader(object):
```

Comment:
This code seems duplicated

Reply:
Do you mean, duplication between koji_promote and koji_upload? I believe the duplication is intentional, while we still maintain the koji_promote plugin (which, IIRC, should be dropped in the future).

Reply:
Yes, I haven't realized that it is koji_promote. ack